src/coreclr/jit/assertionprop.cpp

```diff
@@ -168,6 +168,25 @@ bool IntegralRange::Contains(int64_t value) const
             }
             break;
 
+        case GT_IND:
+        {
+#if defined(_WIN64)
+            if (node->gtGetOp1()->OperGet() == GT_ADD
+                && node->gtGetOp1()->gtGetOp1()->OperGet() == GT_LCL_VAR
+                && node->gtGetOp1()->gtGetOp2()->IsIntegralConst(OFFSETOF__CORINFO_Span__length))
+            {
+                GenTreeLclVar* const lclVar = node->gtGetOp1()->gtGetOp1()->AsLclVar();
+                if (lclVar->IsSpan(compiler))
+                {
+                    // TODO: this check can be moved inside isSpan?
```

Comment:
No, normal locals can be spans too, not just implicit byref params.

Reply:
Thanks